scripts/globals/additional_effects.lua

```diff
@@ -110,209 +111,326 @@ xi.additionalEffect.calcDamage = function(attacker, element, defender, damage)
-            defender:isUndead() or
-            -- Todo: DeathRes has no place in the resistance functions so far..
-            math.random(1, 100) > defender:getMod(xi.mod.DEATHRES) -- We are checking for a fail, not a success.
+            attacker:hasStatusEffect(xi.effect.BLINK) or
+            attacker:hasStatusEffect(xi.effect.COPY_IMAGE)
         then
-            return 0, 0, 0 -- NMs immune or roll failed so return out
+            return 0, 0, 0
         else
-            msgID = xi.msg.basic.ADD_EFFECT_STATUS
-            msgParam = xi.effect.KO
-            defender:setHP(0)
+            attacker:addStatusEffect(xi.effect.BLINK, params.power, 0, params.duration)
+            msgID    = xi.msg.basic.ADD_EFFECT_SELFBUFF
+            msgParam = xi.effect.BLINK
         end
+    elseif params.addStatus == xi.effect.HASTE then
+        attacker:addStatusEffect(xi.effect.HASTE, params.power, 0, params.duration, 0, 0)
+        -- Todo: verify power/duration/tier/overwrite etc
+        msgID    = xi.msg.basic.ADD_EFFECT_SELFBUFF
+        msgParam = xi.effect.HASTE
+    else
+        print('scripts/globals/additional_effects.lua : unhandled additional effect selfbuff! Effect ID: ' .. params.addStatus)
     end
 
-    --[[
-    if msgID == nil then
-        print('Additional effect has a nil msgID !!')
-    elseif msgParam == nil then
-        print('Additional effect has a nil msgParam !!')
+    return subEffect, msgID, msgParam
+end
+
+xi.additionalEffect.procFunctions[xi.additionalEffect.procType.DEATH] =  function(attacker, defender, item, params)
+    local subEffect = params.subEffect
+    local msgID     = 0
+    local msgParam  = 0
+
+    if
+        defender:isNM() or
+        defender:isUndead() or
+        -- Todo: DeathRes has no place in the resistance functions so far..
+        math.random(1, 100) > defender:getMod(xi.mod.DEATHRES) -- We are checking for a fail, not a success.
+    then
+        return 0, 0, 0 -- NMs immune or roll failed so return out
+    else
+        msgID = xi.msg.basic.ADD_EFFECT_STATUS
+        msgParam = xi.effect.KO
+        defender:setHP(0)
     end
-    print('subEffect: '..subEffect..' msgID: '..msgID..' msgParam: '..msgParam)
-    ]]
+
+    return subEffect, msgID, msgParam
+end
+
+xi.additionalEffect.procFunctions[xi.additionalEffect.procType.DEATH] = function(attacker, defender, item, params)
+    local subEffect = params.subEffect
+    local msgID     = 0
+    local msgParam  = 0
+
+    if
+        defender:isNM() or
+        defender:isUndead() or
+        -- Todo: DeathRes has no place in the resistance functions so far..
+        math.random(1, 100) > defender:getMod(xi.mod.DEATHRES) -- We are checking for a fail, not a success.
+    then
+        return 0, 0, 0 -- NMs immune or roll failed so return out
+    else
+        msgID = xi.msg.basic.ADD_EFFECT_STATUS
+        msgParam = xi.effect.KO
+        defender:setHP(0)
+    end
+
     return subEffect, msgID, msgParam
 end
```

Comment:
There are two of these, not sure if this is intended?

Reply:
Whoops, sure isn't

Reply:
fixed